source/gui/addonStoreGui/viewModels/store.py

```diff
@@ -313,22 +347,110 @@ def _handleEnableDisable(self, listItemVM: AddonListItemVM[_AddonManifestModel],
 		listItemVM.status = getStatus(listItemVM.model)
 		self.refresh()
 
-	def enableOverrideIncompatibilityForAddon(self, listItemVM: AddonListItemVM[_AddonManifestModel]) -> None:
+	def enableOverrideIncompatibilityForAddon(
+			self,
+			listItemVM: AddonListItemVM[_AddonManifestModel],
+			askConfirmation: bool = True,
+			useRememberChoiceCheckbox: bool = False,
+	) -> tuple[bool, bool]:
 		from ... import mainFrame
-		if _shouldEnableWhenAddonTooOldDialog(mainFrame, listItemVM.model):
+		if askConfirmation:
+			shouldEnable, shouldRememberChoice = _shouldEnableWhenAddonTooOldDialog(
+				mainFrame,
+				listItemVM.model,
+				
+				useRememberChoiceCheckbox=useRememberChoiceCheckbox,
+			)
+		else:
+			shouldEnable = True
+			shouldRememberChoice = True
+		if shouldEnable:
 			listItemVM.model.enableCompatibilityOverride()
 			self._handleEnableDisable(listItemVM, True)
+		return shouldEnable, shouldRememberChoice
 
 	def enableAddon(self, listItemVM: AddonListItemVM) -> None:
 		self._handleEnableDisable(listItemVM, True)
 
+	def enableAddons(self, listItemVMs: Iterable[AddonListItemVM[_AddonStoreModel]]) -> None:
+		shouldEnableIncompatible = True
+		shouldRememberChoice = False
+		for aVM in listItemVMs:
+			if aVM.canUseEnableOverrideIncompatibilityAction():
+				if shouldRememberChoice:
+					if shouldEnableIncompatible:
+						self.enableOverrideIncompatibilityForAddon(aVM, askConfirmation=False)
+					else:
+						log.debug(
+							f"Skipping {aVM.Id} as override incompatibility has been previously declined for all remaining"
+							" add-ons."
+						)
+				else:
+					shouldEnableIncompatible, shouldRememberChoice = self.enableOverrideIncompatibilityForAddon(
+						aVM,
+						askConfirmation=True,
+						useRememberChoiceCheckbox=True,
+					)
+			elif aVM.canUseEnableAction():
+				self.enableAddon(aVM)
+			else:
+				log.debug(
+					log.debug(f"Skipping {aVM.Id} ({aVM.status}) as it is not relevant for enable action")
+				)
+
+	

```

Comment:
Unnecessary two blank lines here. I have no idea why this has not been caught by Flake8 though.